### PR TITLE
Excise trust state container

### PR DESCRIPTION
**Pull request by daviddahl**
Removal of container usage in contacts (trusted peers)

### CODE REVIEW

**Comment by daviddahl** [comment]
This is a huge bunch of changes that forced me to comment out some integration tests, which, since the session / cookie changes have not worked correctly in PhantomJS, sigh. I plan on changing these tests to run in a real browser in the near future.

File: client/src/account.js [hunk 1]
```diff
@@ -296,8 +296,12 @@ Account.prototype.changePassphrase =
     newKeyring.signKeyPrivateMacSalt = JSON.stringify(signKeyPrivateMacSalt);
     newKeyring.srpVerifier = srpVerifier;
     newKeyring.srpSalt = srpSalt;
-
-    superagent.post(crypton.url() + '/account/' + that.username + '/keyring')
+    var url = crypton.url()
+	  + '/account/'
```
Comment (ecto) [inline client/src/account.js]: indentation

File: client/src/container.js [hunk 2]
```diff
@@ -238,9 +238,11 @@ Container.prototype.getHistory = function (callback) {
   var containerNameHmac = this.getPublicName();
   var currentVersion = this.latestVersion();
 
-  var url = crypton.url() + '/container/' + containerNameHmac + '?after=' + (currentVersion + 1);
+  var url = crypton.url() + '/container/' + containerNameHmac + '?after=' + (currentVersion + 1) + '&sid=' + crypton.sessionId;
+  console.log('getHistory', url);
   superagent.get(url)
     .withCredentials()
+    // .set('X-Session-ID', crypton.sessionId)
```
Comment (ecto) [inline client/src/container.js]: remove commented code

File: client/src/container.js [hunk 3]
```diff
@@ -238,9 +238,11 @@ Container.prototype.getHistory = function (callback) {
   var containerNameHmac = this.getPublicName();
   var currentVersion = this.latestVersion();
 
-  var url = crypton.url() + '/container/' + containerNameHmac + '?after=' + (currentVersion + 1);
+  var url = crypton.url() + '/container/' + containerNameHmac + '?after=' + (currentVersion + 1) + '&sid=' + crypton.sessionId;
+  console.log('getHistory', url);
```
Comment (ecto) [inline client/src/container.js]: remove debug

File: client/src/container.js [hunk 4]
```diff
@@ -377,7 +379,7 @@ Container.prototype.decryptKey = function (record) {
  *
  * @param {Function} callback
  */
-Container.prototype.sync = function (callback) {
+ Container.prototype.sync = function (callback) {
```
Comment (ecto) [inline client/src/container.js]: indentation

File: client/src/core.js [hunk 5]
```diff
@@ -438,10 +445,12 @@ crypton.authorize = function (username, passphrase, callback, options) {
           // calculateSrpM1
           crypton.work.calculateSrpM1(options, function (err, srpM1, ourSrpM2) {
             response = {
-              srpM1: srpM1
+		srpM1: srpM1
```
Comment (ecto) [inline client/src/core.js]: tabs

File: client/src/session.js [hunk 6]
```diff
@@ -41,9 +41,14 @@ var Session = crypton.Session = function (id) {
...
-    secure: true
-  });
+
+  var joinServerParameters = { token: crypton.sessionId };
+  this.socket = io.connect(crypton.url(),
+                           { query: 'joinServerParameters='
+                                  + JSON.stringify(joinServerParameters),
+                             reconnection: true,
+                             reconnectionDelay: 5000
+                           });
```
Comment (ecto) [inline client/src/session.js]: indentation?

File: server/redis-session.js [hunk 7]
```diff
@@ -0,0 +1,288 @@
+// https://github.com/mitchellsimoens/redis-session
```
Comment (ecto) [inline server/redis-session.js]: should be node module

Comment (daviddahl) [inline server/redis-session.js]: Cool. Thanks for looking - keep it coming! Yeah, this should be a module. I need to issue a PR to the original project after fixing one last issue with it.

**Pull request merged**